Adds `Deleted` text to missing resources in JT Details View

AlexSCorey opened the pull request:
##### SUMMARY
This PR addresses https://github.com/ansible/awx/issues/5106
The use-case of this change is if a user deletes an Inventory, or a Project
that is used by a JT they need to know that those resources are missing.

The only time that `Deleted` won't be shown for a missing resource is for
Inventory if it has been marked Prompt on Launch then nothing is shown. in that field.

Also adds icon to indicate that a JT is missing resources on the JT List.
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
7.0.0
```


##### ADDITIONAL INFORMATION
![MissingResources](https://user-images.githubusercontent.com/39280967/67989054-1859ff80-fc08-11e9-83e3-2955f65f1ff2.gif)

AlexSCorey committed:
Adds `Deleted` text to missing resources in JT Detials View

The usecase of this change is if a user deletes an Inventory, or a Project
that is used by a JT they need to know that those resources are missing.

The only time that `Deleted` won't be shown for a missing resource is for
Inventory if it has been marked Prompt on Launch then nothing is shown. in that field.

Also adds icon to indicate that a JT is missing resources on the JT List.

AlexSCorey commented:
recheck

jakemcdermott commented:
recheck

AlexSCorey commented:
recheck

kialam commented on awx/ui_next/src/screens/Template/JobTemplateDetail/JobTemplateDetail.jsx:
```diff
@@ -156,6 +157,28 @@ class JobTemplateDetail extends Component {
+
+    const inventoryValue = (kind, id) => {
+      const inventorykind =
+        kind === 'smart' ? (kind = 'smary_inventory') : (kind = 'inventory');
```
typo here

kialam commented on awx/ui_next/src/components/DetailList/Detail.jsx:
```diff
@@ -34,7 +39,11 @@ const Detail = ({ label, value, fullWidth }) => {
       </DetailName>
-      <DetailValue component={TextListItemVariants.dd} fullWidth={fullWidth}>
+      <DetailValue
+        missingValue={missingValue}
```
I prefer to name boolean props prefixed with `is` to easily denote that I'm expecting either `true` or `false` in the code, e.g. `isMissingValue`.

kialam commented on awx/ui_next/src/components/DetailList/Detail.jsx:
```diff
@@ -23,9 +23,14 @@ const DetailValue = styled(({ fullWidth, ...props }) => (
 `;
 
-const Detail = ({ label, value, fullWidth }) => {
+const Detail = ({ label, value, fullWidth, missingValue }) => {
```
Instead of adding another prop to the `Detail` component for this seldom use case, I think you should use styled components and extend the Detail component to create one for deleted resources.

kialam commented on awx/ui_next/src/screens/Template/JobTemplateDetail/JobTemplateDetail.jsx:
```diff
@@ -156,6 +157,28 @@ class JobTemplateDetail extends Component {
       </TextList>
     );
 
+    const renderMissingDataDetail = value => (
```
I think instead of re-using the `Detail` component and passing it a prop you can just create a `MissingDetail` styled component by extending it:
```
const MissingDetail = styled(Detail)`
  color: #c9190b;
`;
```
And then using it in the `render()` method.

AlexSCorey committed:
Removes missingDetail prop and addresses a typo.

This removes `missingDetail` prop in favor of using styledComponent's className prop.

AlexSCorey committed:
Fixes linting issues

kialam commented on awx/ui_next/src/screens/Template/JobTemplateDetail/JobTemplateDetail.jsx:
```diff
@@ -28,6 +28,12 @@ const ButtonGroup = styled.div`
 `;
+
+const MissingDetail = styled(Detail)`
+  dd& {
```
⭐️
